PavlenkoHW/src/hw20200618_FP/action/ActionEnum.java

```diff
@@ -0,0 +1,14 @@
+package hw20200618_FP.action;
+
+public enum ActionEnum {
+    RUN("Run"),
```

Comment:
The whole idea with this factory is not touch existing code when a new action needs to be added. 

Since the factory is a function, which returns an object by a name, we may consider storing a list of names (strings) somewhere externally, for instance, in a property file. We may keep them in pairs:
`PREVIEW=de.telran.action.PreviewImageAction`

In the factory class we have a static initialization section, where the property file is read, corresponding action classes are instantiated and put to a map, similar to the property file. Then your  `getAction()` method will just lookup in the map and return a pre-initialized action object. May be the enum is not needed then.